linalg/densemat.cpp

```diff
@@ -1718,6 +1718,232 @@ void DenseMatrix::AddMatrix(double a, const DenseMatrix &A, int ro, int co)
    }
 }
 
+void DenseMatrix::GetSubMatrix(const Array<int> & idx, DenseMatrix & A) const
+{
+   int k = idx.Size();
+   A.SetSize(k);
```

Comment:
We should add MFEM_VERIFY (or similar) checks to confirm that A and this->data won't be indexed out of range.

Reply:
done!